src/lib/kadm5/srv/svr_principal.c

```diff
@@ -1437,12 +1442,9 @@ kadm5_chpass_principal_3(void *server_handle,
             }
         }
 
+        kdb->mask |= KADM5_PW_EXPIRATION;
```

Comment:
I think this belongs before the if statement, since in chpass we always want to change pw_expiration.